frontends/web/src/components/progressRing/progressRing.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2019 Shift Devices AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { h, RenderableProps } from 'preact';
+import * as style from './progressRing.css';
+
+interface ProgressRingProps {
+    width: number;
+    value: number;
+    className?: string[] | string;
+    generic?: boolean;
+    isComplete?: boolean | undefined;
+    isError?: boolean;
+}
+
+const ProgressRing = ({
+    className,
+    generic,
+    isComplete,
+    isError,
+    value,
+    width,
+}: RenderableProps<ProgressRingProps>) => {
+    const radius = (width - 2) / 2;
+    const circumference = radius * 2 * Math.PI;
+    const progress = isError ? 100 : value / 100;
+    const offset = circumference * (1 - progress);
+    return (
+        <svg
+            className={[style.container, className ? className : ''].join(' ')}
+            width={width}
+            height={width}
+            viewBox={`0 0 ${width} ${width}`}>
+            <circle
+                className={style.background}
+                cx={width / 2}
+                cy={width / 2}
+                r={radius}
+                strokeWidth={2} />
```

Comment:
I tested this PR on a low res screen and the circles were a bit hard to see, especially the gray/pending (Elvis will complain I'm sure :D)
I tried changing the strokeWidth to 3 and it looks quite nice. What do you think?

Reply:
sure, let's give it a shot and we can adjust if any other opinions/better ideas come up. thanks